cunumeric/random/random.py

```diff
@@ -216,6 +217,96 @@ def chisquare(
     return generator.get_static_generator().chisquare(df, size, dtype)
 
 
+def choice(
+    a: Union[int, ndarray],
+    size: Union[NdShapeLike, None] = None,
+    replace: bool = True,
+    p: Union[None, ndarray] = None,
+) -> Union[int, ndarray, npt.NDArray[Any]]:
+    """
+    Returns an array of random values from a given 1-D array.
+    Each element of the returned array is independently sampled
+    from ``a`` according to ``p`` or uniformly.
+
+    Note
+    ----
+       Currently ``p`` is not supported when ``replace=False``.
+
+    Parameters
+    ----------
+    a : ndarray or int : If an array-like,
+            a random sample is generated from its elements.
+            If an int, the random sample is generated as if ``a`` was
+            ``np.arange(n)``
+    size : int or tuple of ints : The shape of the array.
+    replace (boolean): Whether the sample is with or without replacement.
+    p : 1-D array-like :
+            The probabilities associated with each entry in ``a``.
+            If not given the sample assumes a uniform distribution over all
+            entries in ``a``.
```

Comment:
optional params need "optional" and also to state the default values. c.f. numpy docstring:

https://numpy.org/doc/stable/reference/random/generated/numpy.random.choice.html